MapboxNavigation/NavigationMapView.swift

```diff
@@ -306,6 +307,8 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         } else {
             preferredFramesPerSecond = FrameIntervalOptions.pluggedInFramesPerSecond
         }
+        
+        updateCourseTracking(location: location, animated: true)
```

Comment:
This method is already risky and could become moreso if we rely on notifications to drive course updates as opposed to manual calls to `updateCourseTracking(location:animated:)`. NavigationMapView apparently listens for notifications coming from _any_ RouteController, so every map view in the application will automatically respond to every route controller in the application. There’s a high potential for crosstalk. We’d need to pass in a specific route controller as the `object` here to avoid that problem:

https://github.com/mapbox/mapbox-navigation-ios/blob/ea468da75a44cfd2153b2071968e35f2765809b7/MapboxNavigation/NavigationMapView.swift#L274

Reply:
@1ec5 this may require changing the initializer on NavigationMapview. We currently have no `RouterController` props to grab.

Reply:
I guess making this an optional property is another option. If you only want to update a single map view, set this prop.

Reply:
When we implemented #402, the idea was that moving the course view was the responsibility of the view controller that contains the map view, not the map view itself, since the view controller knows about its route controller for sure. I think we could’ve made that more clear by making the course view a sibling of the map view instead of one of its views. Relying on a global notification is a poor way to make the map view aware of the view controller’s route controller. Changing the initializer is also problematic because map views are often created using the `init(frame:)` or `init(coder:)` initializer.

> If you only want to update a single map view, set this prop.

I don’t think it should be something that you only use in some cases. If we make the route controller a property, then the view controller should always have to set this property on the map view in order to keep the course view in sync. Otherwise, how would the map view handle the route controller swapping that happens around tunnels?

Reply:
How would someone add a NavigationMapView to a storyboard if they have not yet created a route controller to pass through?